tests/core/test_funcs.py

```diff
@@ -0,0 +1,68 @@
+import uuid
+from typing import Optional
+
+import pytest
+
+from dls_bluesky_core.core import group_uuid, in_micros, step_to_num
+
+
+@pytest.mark.parametrize(
+    "s,us",
+    [
+        (4.000_001, 4_000_001),
+        (4.999_999, 4_999_999),
+        (4, 4_000_000),
+        (-4.000_001, -4_000_001),
+        (-4.999_999, -4_999_999),
+        (-4, -4_000_000),
+        (4.000_000_1, 4_000_001),
+        (4.999_999_9, 5_000_000),
+        (0.1, 100_000),
+        (0.000_000_1, 1),
+        (-4.000_000_5, -4_000_000),
+        (-4.999_999_9, -4_999_999),
+        (-4.05, -4_050_000),
```

Comment:
@coretl Or should these round away from 0?

Reply:
We should raise on a negative time